backend/app/glossary/controllers.py

```diff
@@ -19,4 +20,15 @@ def create_glossary_doc_from_file_controller(
 
 
 def list_glossary_docs_controller(db: Session):
-    return GlossaryDocsQuery(db).list_glossary_docs()
+    glossaries = GlossaryDocsQuery(db).list_glossary_docs()
+    return [
+        GlossaryDocumentResponse.model_validate(glossary) for glossary in glossaries
+    ]
+
+
+def retrieve_glossary_doc_controller(glossary_doc_id: int, db: Session):
+    try:
+        doc = GlossaryDocsQuery(db).get_glossary_doc(glossary_doc_id)
+        return GlossaryDocumentResponse.model_validate(doc)
```

Comment:
What is the purpose of model validation here after requesting it from DB?

Reply:
Validate a pydantic model instance
https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel.model_validate

Cast from DbModel instance to Pydantic instance